test/pinning_test.jl

```diff
@@ -132,3 +132,16 @@ end
     pinthreads(reverse(cpuids); force = false)
     @test getcpuids() == cpuids
 end
+
+@testset "Thread Pinning (random)" begin
+    pinthreads(0:nthreads() - 1)
```

Comment:
Would be great to generalise this and not explicitly assume that the cores are numbered 0, 1, 2, ... (for example, on Fugaku this assumption is wrong). You can use the function `cpuids_all` to get all valid cpu ids on the current system.

Reply:
Interesting, I didn't realize that :thinking: Fixed in e54f276.